provider/src/cache.ts

```diff
@@ -68,13 +69,65 @@ export const storeRPCResponseInCache = async (
   env: Env
 ): Promise<boolean> => {
   try {
-    // Check if this is a cacheable request
+    // Handle batch requests by storing each response individually
+    if (Array.isArray(body) && Array.isArray(json)) {
```

Comment:
A better aporoach here is just to make sure this function is always called with an array?

Reply:
so we mive the logic of handling batch or unique at a higher level?